apps/air-discount-scheme/backend/src/app/modules/user/user.controller.ts

```diff
@@ -52,19 +60,29 @@ export class PublicUserController {
   }
 }
 
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes('@vegagerdin.is/air-discount-scheme-scope')
 @Controller('api/private')
 export class PrivateUserController {
   constructor(
     private readonly flightService: FlightService,
     private readonly userService: UserService,
+    private thjodskraXroad: NationalRegistryXRoadService,
   ) {}
 
   @Get('users/:nationalId/relations')
   @ApiExcludeEndpoint()
   async getUserRelations(
     @Param() params: GetUserRelationsParams,
+    @CurrentUser() user: AuthUser,
   ): Promise<User[]> {
-    const relations = await this.userService.getRelations(params.nationalId)
+    const relations = await this.thjodskraXroad.getChildrenForsja(
+      user,
+      params.nationalId,
```

Comment:
Why don't you use the national id from the user instead of using params here?

Don't know the flow but it feels like more sanitized.

Reply:
I originally did this because I had access to params.nationalId, meaning I could verify that the nationalId from currentUser-authNestTools would have the same nationalid as was sent into the function. This line of thought was only possible because nationalRegistry would throw error if user.authorization !== nationalId

Reply:
Is the UI only sending the nationalId of the currently logged in user? If so I agree with Rikki about removing the params and read the nationalId from the current user.

Reply:
the params.nationalId is taken from ADS-API currentUser.decorator from cookies.